include/libdnf/conf/vars.hpp

```diff
@@ -34,19 +34,43 @@ namespace libdnf {
 /// of directories.
 struct Vars {
 public:
+    enum class Priority { DEFAULT = 10, AUTO = 20, FILE = 30, ENVIRONMENT = 40, COMMANDLINE = 50, RUNTIME = 60 };
+
+    struct Variable {
+        std::string value;
+        Priority priority;
+    };
+
     /// @brief Substitute DNF vars in the input text.
     ///
     /// @param text The text for substitution
     /// @return The substituted text
     std::string substitute(const std::string & text) const;
 
-    const std::map<std::string, std::string> & get_values() const { return values; }
+    const std::map<std::string, Variable> & get_variables() const { return variables; }
 
     /// @brief Set particular variable to a value
     ///
     /// @param name Name of the variable
     /// @param value Value to be stored in variable
-    void set_value(const std::string & name, const std::string & value) { values[name] = value; }
+    /// @param prio Source/Priority of the value
+    void set(const std::string & name, const std::string & value, Priority prio = Priority::RUNTIME);
+
+    /// @brief Checks if there is an variable with name equivalent to name in the container.
+    ///
+    /// @param name Name of the variable
+    /// @return true if there is such an element, otherwise false
+    bool contains(const std::string & name) const { return variables.find(name) != variables.end(); }
+
+    /// @brief Get value of particular variable.
+    ///
+    /// @param name Name of the variable
+    const std::string & get_value(const std::string & name) const { return variables.at(name).value; }
```

Comment:
Am I correct that it raises an exception when variable is not there?

Reply:
Please could you mention that it raises an exception? The question is how we suppose to use it? When I have a map I prefer to use `find()` and then from iterator value is not `.end()`. Also I can use `.at()` but it requires handling of exceptions. In our API we have `contains` and `get_value`. Using at beginning `contains` and then `get_value` is not efficient. Do we suggest to use `get_value()` with `catch` as the best approach?

Reply:
The comment is also valid for `get()`